crates/ide-assists/src/handlers/auto_import.rs

```diff
@@ -272,8 +272,10 @@ fn module_distance_heuristic(db: &dyn HirDatabase, current: &Module, item: &Modu
     // cost of importing from another crate
     let crate_boundary_cost = if current.krate() == item.krate() {
         0
-    } else if item.krate().is_builtin(db) {
+    } else if item.krate().origin(db).is_local() {
```

Comment:
Does this pick up examples / binaries defined in the crate?

Reply:
it should. `is_local` generally means “everything that can reasonably be edited by the user/isn’t a crate downloaded from crates.io”.

Reply:
> Does this pick up examples / binaries defined in the crate?

Are you asking if it picks up _items that are defined in_ examples / binaries from library code? If so, correct me if I'm wrong, but I don't think you can do that in Rust?

If you mean the opposite, yes this does that. `is_local` specifically checks if the crate origin is local to the workspace.

Reply:
I meant the opposite (picking up library defined types from the example code)

Reply:
examples, binaries, library and integration tests of a package are all different crates, so the `==` condition above will fail for those. `is_local` is about where the package is coming from, local being its a workspace member (opposed to coming from crates-io)

Reply:
Checking whether things are from the same package can't really be done (we discard that informaton). We could specifically try to rank better by checking if the item crate is in the direct dependencies of the current crate though, that will make things work fine for examples and the like